### DISCUSSION TITLE
POA and loss "conventions"

### DISCUSSION THREAD

**Discussion by williamhobbs** [williamhobbs/2024_pvpmc_self_shade#3] [General]
@kandersolar In "rev2" of the notebook, I changed up a few key things:

1. I have separate calculations for total frontside POA with and without impacts of direct shade from adjacent rows:
```python
    # total poa on the front, but without direct shade impacts (keeping diffuse impacts from infinite_sheds)
    poa_front_total_without_direct_shade = irrad_inf_sh['poa_front_diffuse'] + poa_front_direct_unshaded
```
 and 
```python
        # total POA on the front *with* direct shade impacts for each course/string
        poa_front_total_with_direct_shade[n] = irrad_inf_sh['poa_front_diffuse'] + ((1-fs[n]) * poa_front_direct_unshaded)
```
2. Then diffuse fraction for each course/string is based on the POA with direct shade for the course/string:
```python
        # diffuse fraction for each course/string
        fd[n] = irrad_inf_sh['poa_front_diffuse'] / poa_front_total_with_direct_shade[n]
```
3. I now apply the `shade_loss` for each course/string to the POA without direct shade impacts before passing to `pvlib.pvsystem.pvwatts_dc`.
```python
        # adjust irradiance based on modeled shade loss
        poa_effective = (1 - shade_loss[n]) * poa_front_total_without_direct_shade
```

The comparisons with observations look better to me, particularly in the 30 minutes or so closest to sunrise/sunset. 

It's possible that I've thought myself in circles and simply undid something that we agreed needed to be there, but I _think_ this all makes sense.

**Comment by kandersolar** [comment]
For the diffuse fraction / shading loss part, I think there is an error.  Consider what happens in the hypothetical extreme of shaded_fraction=100%.  Then all of the beam irradiance will be shaded, and so the calculation of `fd[n] = irrad_inf_sh['poa_front_diffuse'] / poa_front_total_with_direct_shade[n]` will give a diffuse fraction of 100%.  That means that the shade loss will be calculated as 0%.

I think instead it should be calculated as `fd[n] = irrad_inf_sh['poa_front_diffuse'] / poa_front_total_without_direct_shade`.

BTW I'm curious how much the results change if you switch to a better clear-sky model, for example PSM3.

**Reply by williamhobbs** [reply]
Hmmm... good point. I think that puts us back where we were, but maybe with better variable names this time. 

>BTW I'm curious how much the results change if you switch to a better clear-sky model, for example PSM3.

I thought about that, too. Might result in more diffuse light at low solar elevation angles.

**Comment by williamhobbs** [comment]
I added NSRDB for plant40/41 in rev2. I think it looks much better:

[edit - fixed the legend]
![image](https://github.com/williamhobbs/2024_pvpmc_self_shade/assets/45701090/8790961c-c231-4b9f-9fae-eee38d047bbb)

Using `loc.get_clearsky()`:

![image](https://github.com/williamhobbs/2024_pvpmc_self_shade/assets/45701090/5d6f2246-8e52-4811-b3dd-97af92fd7ec4)
